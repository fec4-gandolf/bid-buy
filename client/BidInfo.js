import React from 'react';
import styled from 'styled-components';

const Td = styled.td`
  padding: 1em;
`;

const MiddleRow = styled.tr`
  background: #e2e2e2;
  vertical-align: top;
`;

const Span = styled.span`
  font-size: small;
  color: #666;
`;


const BidButton = styled.button`
  background: #0068af;
  color: #fff
  font-weight: bold;
  display: block;
  border-radius: 5px;
  font-size: 13pt;
  width: 128px;
  padding: 5px 0;
  margin: 1em 0;
  :active {
    position: relative;
    top: 1px;
  }
`;

class BidInfo extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      bid: ''
    };
  }

  render() {
    const product = this.props.product;

    return (
      <tbody>
        <MiddleRow>
          <Td>Current Bid:</Td>
          <Td>
            US ${product.price.toFixed(2)}
            <form>
              <input type="text"></input> <br />
              <Span>Enter US ${(product.price + 1).toFixed(2)} or more</Span>
            </form>
          </Td>
          <Td>
            [ <a>{product.bids} bids</a> ]
            <BidButton>Place bid</BidButton>
          </Td>
        </MiddleRow>
      </tbody>
    );
  }
}

export default BidInfo;