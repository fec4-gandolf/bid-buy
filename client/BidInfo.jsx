import React from 'react';
import styled from 'styled-components';

const Td = styled.td`
  padding: 1em;
`;

const MiddleRow = styled.tr`
  background: #e2e2e2;
  vertical-align: top;
`;

const Span = styled.span`
  font-size: small;
  color: #666;
`;

const BidButton = styled.button`
  background: #0068af;
  color: #fff
  font-weight: bold;
  display: block;
  border-radius: 5px;
  font-size: 13pt;
  width: 128px;
  padding: 5px 0;
  margin: 1em 0;
  :active {
    position: relative;
    top: 1px;
  }
`;

class BidInfo extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      bid: ''
    };

    this.handleFormChange = this.handleFormChange.bind(this);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  handleFormChange(e) {
    this.setState({
      bid: e.target.value
    });
  }

  handleFormSubmit(e) {
    e.preventDefault();
    this.props.placeBid(this.state.bid);
    this.setState({
      bid: ''
    });
  }

  render() {
    const product = this.props.product;

    return (
      <tbody>
        <MiddleRow>
          <Td>Current Bid:</Td>
          <Td>
            US ${product.price.toFixed(2)}
            <form onSubmit={this.handleFormSubmit}>
              <input
                type="text"
                name="bid"
                value={this.state.bid}
                onChange={this.handleFormChange}
              ></input> <br />
              <Span>Enter US ${(product.price + 1).toFixed(2)} or more</Span>
            </form>
          </Td>
          <Td>
            [ <a>{product.bids} bids</a> ]
            <BidButton onClick={this.handleFormSubmit}>Place bid</BidButton>
          </Td>
        </MiddleRow>
      </tbody>
    );
  }
}

export default BidInfo;
