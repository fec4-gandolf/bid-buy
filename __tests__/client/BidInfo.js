import React from 'react';
import BidInfo from '../../client/BidInfo';
import renderer from 'react-test-renderer';
import placeholderProduct from './placeholderProduct';

it('renders correctly', () => {
  const tree = renderer
    .create(<BidInfo product={placeholderProduct} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
