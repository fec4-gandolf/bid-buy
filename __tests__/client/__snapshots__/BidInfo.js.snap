// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tbody>
  <tr
    className="sc-bwzfXH hynJvh"
  >
    <td
      className="sc-bdVaJa bYMmDY"
    >
      Current Bid:
    </td>
    <td
      className="sc-bdVaJa bYMmDY"
    >
      US $
      564.00
      <form
        onSubmit={[Function]}
      >
        <input
          name="bid"
          onChange={[Function]}
          type="text"
          value=""
        />
         
        <br />
        <span
          className="sc-htpNat eMRilx"
        >
          Enter US $
          565.00
           or more
        </span>
      </form>
    </td>
    <td
      className="sc-bdVaJa bYMmDY"
    >
      [ 
      <a>
        37
         bids
      </a>
       ]
      <button
        className="sc-bxivhb bLfBYJ"
        onClick={[Function]}
      >
        Place bid
      </button>
    </td>
  </tr>
</tbody>
`;
